packages/lib/src/application/Package.ts

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash';
+
+import Logger from '../utils/Logger';
+import Contracts from '../artifacts/Contracts';
+import ImplementationDirectory from '../application/ImplementationDirectory';
+import { toSemanticVersion, SemanticVersion } from '../utils/Semver';
+import { toAddress, isZeroAddress } from '../utils/Addresses';
+import ContractFactory, { ContractWrapper } from '../artifacts/ContractFactory';
+import { deploy as deployContract, sendTransaction } from '../utils/Transactions';
+
+const log: any = new Logger('Package');
+
+interface PackageInterface extends ContractWrapper {
+  address: string;
+  hasVersion?;
```

Comment:
Probably an boolean: `hasVersion?:boolean;`

Reply:
`hasVersion` is a function. I prefer leaving this as it is, as we are probably adding `TypeChain` or whatsoever to type the ABI.